### ISSUE TITLE
funcName not in scope

### ISSUE THREAD

**Issue by apatil** [purcell/servant-elm#7]
Hello,

I'm trying to build servant-elm with the GitHub heads of servant and elm-export. I'm getting the following:

```
[1 of 1] Compiling Servant.Elm      ( src/Servant/Elm.hs, dist/build/Servant/Elm.o )

src/Servant/Elm.hs:90:44: Not in scope: ‘funcName’

src/Servant/Elm.hs:103:23:
    Not in scope: ‘segment’
    Perhaps you meant one of these:
      data constructor ‘Segment’ (imported from Servant.Foreign),
      ‘segments’ (line 116)

src/Servant/Elm.hs:111:19:
    Not in scope: ‘segment’
    Perhaps you meant one of these:
      data constructor ‘Segment’ (imported from Servant.Foreign),
      ‘segments’ (line 116)
```

I'm guessing these are imported from servant-foreign, but neither funcName nor segment occurs in the current github head of servant.

Which commit of servant are you building against?

Thanks!

**Comment by purcell** [comment]
The upstream commit is specified in `stack.yml` IIRC, so you can't expect to just build against the latest upstream code: unfortunately there isn't a working version of `servant-elm` yet. In order to generate useful type-safe Elm client code I'm waiting on upstream changes to `servant-foreign`, e.g. those I have proposed in https://github.com/haskell-servant/servant/pull/351, or a better implementation of those by the servant maintainers themselves.

**Comment by purcell** [comment]
I'll close this issue for now, but I'm grateful for your interest in `servant-elm`, and I'll try to ping you when I make some progress -- I think it's going to be a really useful too. :-)

**Comment by apatil** [comment]
Nice, I did get it built with that commit. Looking forward to seeing what you come up with!
